src/globals/js/mixins/README.md

```diff
@@ -0,0 +1,201 @@
+# Carbon component mixins
+
+Carbon component mixins, based on [Subclass Factory Pattern](https://github.com/justinfagnani/proposal-mixins#subclass-factory-pattern), provides the basis for Carbon component classes by allowing component implementation to compose small pieces of functionalities to base them on, instead of introducing "fat base class".
+
+* [Component lifecycle](#component-lifecycle-create-componentjs) ([`create-component.js`](./create-component.js))
```

Comment:
You could use [`doctoc`](https://github.com/thlorenz/doctoc) here to generate the table of contents if you wanted 😄 

Used in DataTable in the README: https://raw.githubusercontent.com/carbon-design-system/carbon-components-react/master/src/components/DataTable/README.md

The snippet itself just looks like:

```md
<!-- To run doctoc, just do `npx doctoc README.md` in this directory! -->

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->


- [Installation](#installation)
- [Usage](#usage)
- [Props](#props)
  - [`rows`](#rows)
  - [`headers`](#headers)
  - [`sortRow`](#sortrow)
  - [`filterRows`](#filterrows)
  - [`locale`](#locale)
- [Render Prop Function](#render-prop-function)
  - [Prop Getters](#prop-getters)
  - [Actions](#actions)
  - [State](#state)
  - [Props](#props-1)
- [Use-cases](#use-cases)
  - [Sorting](#sorting)
    - [Programmatic sorting](#programmatic-sorting)
    - [Custom sorting](#custom-sorting)
  - [Expansion](#expansion)
    - [Programmatic expansion](#programmatic-expansion)
  - [Selection](#selection)
    - [Programmatic selection](#programmatic-selection)
  - [Filtering](#filtering)
  - [Batch Actions](#batch-actions)
- [Attribution](#attribution)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->
```

Reply:
Done, thanks @joshblack for suggesting!